src/UglyToad.PdfPig/Export/HOcrTextExporter.cs

```diff
@@ -0,0 +1,358 @@
+﻿using System;
+using System.Linq;
+using UglyToad.PdfPig.Content;
+using UglyToad.PdfPig.DocumentLayoutAnalysis;
+using UglyToad.PdfPig.Geometry;
+using UglyToad.PdfPig.Util;
+
+namespace UglyToad.PdfPig.Export
+{
+    /// <summary>
+    /// hOCR v1.2 (HTML) text exporter.
+    /// <para>See http://kba.cloud/hocr-spec/1.2/ </para>
+    /// </summary>
+    internal class HOcrTextExporter : ITextExporter
+    {
+        private const string xmlHeader = "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n<!DOCTYPE html PUBLIC \"-//W3C//DTD XHTML 1.0 Transitional//EN\"\n\t\"http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd\">\n";
+        private const string hocrjs = "<script src='https://unpkg.com/hocrjs'></script>\n";
+
+        private IPageSegmenter pageSegmenter;
+        private IWordExtractor wordExtractor;
+
+        private decimal scale;
+        private string indentChar;
+
+        private int pageCount = 0;
+        private int areaCount = 0;
+        private int lineCount = 0;
+        private int wordCount = 0;
+        private int pathCount = 0;
+        private int paraCount = 0;
+        private int imageCount = 0;
+
+        /// <summary>
+        /// hOCR v1.2 (HTML)
+        /// <para>See http://kba.cloud/hocr-spec/1.2/ </para>
+        /// </summary>
+        /// <param name="wordExtractor"></param>
+        /// <param name="pageSegmenter"></param>
+        /// <param name="scale"></param>
+        /// <param name="indent">Indent character.</param>
+        public HOcrTextExporter(IWordExtractor wordExtractor, IPageSegmenter pageSegmenter, double scale = 1.0, string indent = "\t")
+        {
+            this.wordExtractor = wordExtractor;
+            this.pageSegmenter = pageSegmenter;
+            this.scale = (decimal)scale;
+            this.indentChar = indent;
+        }
+
+        /// <summary>
+        /// Get the hORC (HTML) string of the page layout.
+        /// </summary>
+        /// <param name="document">The document.</param>
+        /// <param name="includePaths">Draw <see cref="PdfPath"/>s present in the page.</param>
+        /// <param name="useHocrjs">Will add a reference to the 'hocrjs' script just before the closing 'body' tag, adding the 
+        /// interface to a plain hOCR file.<para>See https://github.com/kba/hocrjs for more information.</para></param>
+        public string Get(PdfDocument document, bool includePaths = false, bool useHocrjs = false)
+        {
+            string hocr = GetHead() + indentChar + "<body>\n";
```

Comment:
I'm a bit worried about using string concatenation to generate XML, it seems like something that's easy to get wrong, it's probably OK as a first pass but I'm not sure how well it'll handle escape characters etc. I'd look into either using `XDocument` or `XmlSerializer` for this use case too, I know it's HTML so technically different but I think it's better to use something like that to generate well-formatted XML.

Reply:
Hi Eliot, agreed on this, but I have not found the hOCR schema, so I had to do it manually. This is the approach used in Tesseract as well for hOCR. See https://github.com/tesseract-ocr/tesseract/blob/master/src/api/hocrrenderer.cpp

If you are able to find the Schema, happy to look at it 😄